src/Native/System.Native/pal_networking.cpp

```diff
@@ -67,13 +67,12 @@ static void ConvertByteArrayToSockAddrIn(sockaddr_in& addr, const uint8_t* buffe
     addr.sin_family = AF_INET;
 }
 
-static void ConvertSockAddrInToByteArray(uint8_t* buffer, int32_t bufferLength, const sockaddr_in& addr)
+static void ConvertInAddrToByteArray(uint8_t* buffer, int32_t bufferLength, const in_addr& addr)
 {
     assert(bufferLength == NUM_BYTES_IN_IPV4_ADDRESS);
     (void)bufferLength; // Silence compiler warnings about unused variables on release mode
```

Comment:
I think we should fail in release for anything that would overflow. I'm tempted to add a PAL_REQUIRE macro that falis fast in release, but so far I've spelled out the checks with proper `if`s and returned error codes. You can leave this since it predates this particular change, but using this as an opportunity to discuss the direction we want to take for these sorts of checks. @stephentoub, @sokket thoughts?